src/amplify-migration/amplifyMigrationScript.ts

```diff
@@ -344,51 +362,84 @@ export async function changeFileContent({
       fileContent = fileContent.replace(match, newMatch);
     }
 
-  
-    const { fileContent: filepathContent } =
-      await updateFilesUploadsPath(
-        url,
-        setOfAllDocumentsPath,
-        currentRepoName
-      );
+    const { fileContent: filepathContent } = await updateFilesUploadsPath(
+      url,
+      setOfAllDocumentsPath,
+      currentRepoName
+    );
     fileContent = fileContent.replace(url, filepathContent);
   }
 
-  return { fileContent, };
+  const yamlParser = YAML.parseAllDocuments(fileContent);
+
+  /**
+   * This is to handle the case where the yml file has multiple documents which are
+   * separated by document end marker lines. Since we don't expect to have > 1 yml
+   * document in a single file, we will only process the first document. The other
+   * documents in this array are expected to be null.
+   */
+  const yamlDocument = yamlParser[0];
+
+  /**
+   * This is a safe cast since we our yaml files are all This represents a YAML mapping,
```

Comment:
The first sentence of the comment seems disjointed haha

Reply:
86766bcc12d1bf6a669b641e7277fcc755951038